zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,697 @@
+        if deltas is not None:
+            deltas = bz.transform(
+                deltas,
+                **{TS_FIELD_NAME: deltas[AD_FIELD_NAME]}
+            )
+    else:
+        _check_datetime_field(TS_FIELD_NAME, measure)
+
+    if deltas is not None and (sorted(deltas.dshape.measure.fields) !=
+                               sorted(measure.fields)):
+        raise TypeError(
+            'base measure != deltas measure:\n%s != %s' % (
+                measure,
+                deltas.dshape.measure,
+            ),
+        )
+
+    # Ensure that we have a data resource to execute the query against.
+    _check_resources('expr', expr, resources)
+    _check_resources('deltas', deltas, resources)
+
+    # Create or retrieve the pipeline api dataset.
+    ds = new_dataset(expr, deltas)
+    # Register our new dataset with the loader.
+    (loader if loader is not None else global_loader)[ds] = ExprData(
+        expr,
+        deltas,
+        resources,
+    )
+    if single_column is not None:
+        # We were passed a single column, extract and return it.
+        return getattr(ds, single_column)
+    return ds
+
+
+getdataset = attrgetter('dataset')
+dataset_name = attrgetter('name')
+
+
+def inline_novel_deltas(base, deltas, dates):
+    """Inline any deltas into the base set that would have changed our most
+    recently known value.
+
+    Parameters
+    ----------
+    base : pd.DataFrame
+        The first known values.
+    deltas : pd.DataFrame
+        Overwrites to the base data.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+
+    Returns
+    -------
+    new_base : pd.DataFrame
+        The new base data with novel deltas inserted.
+    """
+    get_indexes = dates.searchsorted
+    return pd.concat(
+        (base,
+         deltas.loc[
+             (get_indexes(deltas[TS_FIELD_NAME].values, 'right') -
+              get_indexes(deltas[AD_FIELD_NAME].values, 'letf')) <= 1
+         ].drop(AD_FIELD_NAME, 1)),
+        ignore_index=True,
+    )
+
+
+def overwrite_from_dates(asof, dates, dense_dates, asset_idx, value):
+    """Construct a `Float64Overwrite` with the correct
+    start and end date based on the asof date of the delta,
+    the dense_dates, and the dense_dates.
+
+    Parameters
+    ----------
+    asof : datetime
+        The asof date of the delta.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    dense_dates : pd.DatetimeIndex
+        The dates that appeared in the dataset.
+    asset_idx : int
+        The index of the asset in the block.
+    value : np.float64
+        The value to overwrite with.
+
+    Returns
+    -------
+    overwrite : Float64Overwrite
+        The overwrite that will apply the new value to the data.
+    """
+    return Float64Overwrite(
+        dates.searchsorted(asof),
+        dates.get_loc(dense_dates[dense_dates.searchsorted(asof) + 1]) - 1,
```

Comment:
The +1 and -1 set off alarms for me.  What is this trying to do?  Is it meaningfully different from `get_adjustment_locs` in adjustment.pyx?